packages/experimental/Switch/src/Switch.win32.tsx

```diff
@@ -36,26 +37,80 @@ export const Switch = compose<SwitchType>({
+    const [textBeingTested, setTextBeingTested] = React.useState<textBeingTestedStates>('init');
+    const { onText, offText } = userProps;
+    const toggleContainerRef = React.useRef(null);
     const switchInfo = useSwitch(userProps);
+
     // grab the styled slots
     const Slots = useSlots(userProps, (layer) => switchLookup(layer, switchInfo.state, switchInfo.props));
 
+    /*
+      Controls the rendering of the onText and offText in order to take measurements of what the minWidth should
+      be. Having the correct minWidth prevents the control from "jiggling" when the onText and offText
+      are different sizes.
+    */
+    const measureOnOffTexts = React.useCallback(
+      (measuredWidth) => {
+        switch (textBeingTested) {
+          case 'init':
+            setTextBeingTested('onText');
+            setOnOffTextTest(onText);
+            break;
+          case 'onText':
+            setFinalWidth(0);
+            setTextBeingTested('offText');
+            setOnOffTextTest(offText);
+            break;
+          case 'offText':
+            setFinalWidth(measuredWidth);
+            setTextBeingTested('done');
+            setOnOffTextTest(null);
+            break;
+          case 'done':
+            if (measuredWidth > finalWidth) {
+              setFinalWidth(measuredWidth);
+            }
+            break;
+        }
+      },
+      [textBeingTested, finalWidth, offText, onText],
+    );
+
+    // Each time we reach a new stage in the measureOnOffTexts method we take a measurement.
+    React.useEffect(() => {
+      toggleContainerRef?.current?.measure((_x, _y, width, _height) => {
+        measureOnOffTexts(width);
+      });
+    }, [textBeingTested, measureOnOffTexts]);
+
+    // If the user changes the either the onText or offText props we trigger a re-measurement of the onText and offText.
+    React.useEffect(() => {
+      if (textBeingTested === 'done') {
+        setTextBeingTested('init');
+      }
+    }, [onText, offText]);
+
     // now return the handler for finishing render
     return (final: SwitchProps) => {
       const { label, offText, onText, labelPosition, ...mergedProps } = mergeProps(switchInfo.props, final);
       const onOffText = switchInfo.state.toggled ? onText : offText;
       const displayOnOffText = !!offText || !!onText;
       const isReduceMotionEnabled = AccessibilityInfo.isReduceMotionEnabled;
       const thumbAnimation = isReduceMotionEnabled ? { animationClass: 'Ribbon_SwitchThumb' } : null;
+      const currentOpacity = onOffTextTest ? 0 : 1;
+      const newMinWidth = onOffTextTest ? null : { minWidth: finalWidth };
 
       return (
-        <Slots.root {...mergedProps}>
+        <Slots.root {...mergedProps} style={{opacity: currentOpacity}}>
           <Slots.label>{label}</Slots.label>
-          <Slots.toggleContainer>
+          <Slots.toggleContainer ref={toggleContainerRef} style={newMinWidth}>
```

Comment:
>style={newMinWidth} [](http://example.com/codeflow?start=58&length=19)

Would this stomp any styling that the toggleContainer has on it from buildSlotProps/the styling file?

Reply:
I've tested it and it looks like it doesn't.